snyk/models.py

```diff
@@ -624,6 +623,76 @@ def move(self, new_org_id: str) -> bool:
+        """
+        project_snapshot_result = self.organization.client.post(
+            f"org/{self.organization.id}/project/{self.id}/history?perPage=1&page=1",
+            {},
+        )
+        return project_snapshot_result.json().get("snapshots", [{}])[0]
+
+    def __getattr__(self, item):
+        """
+        Will handle lazy loading of attributes which require further API calls or are computationally expensive. This
+        avoids having to load them when we retrieve the full list from the API.
+        """
+        # These attributes require us to get the latest snapshot
+        if item in [
+            "totalDependencies",
+            "lastTestedDate",
+            "imageId",
+            "imageTag",
+            "imageBaseImage",
+            "imagePlatform",
+        ]:
+            snapshot = self._get_project_snapshot()
+            if item == "totalDependencies":
+                return snapshot.get("totalDependencies", 0)
+            elif item == "lastTestedDate":
+                return snapshot.get("created")
+            elif item == "imageId":
+                return snapshot.get("imageId")
+            elif item == "imageTag":
+                return snapshot.get("imageTag")
+            elif item == "imageBaseImage":
+                return snapshot.get("baseImageName")
+            elif item == "imagePlatform":
+                return snapshot.get("imagePlatform")
+        # These attributes require us to call the user API to get a users details
+        elif item in ["importingUser", "owner"]:
+            if item == "importingUser":
+                selected_user = self.importingUserId
+            else:
+                selected_user = self.owningUserId
+            user_response = self.organization.client.get(
+                f"orgs/{self.organization.id}/users/{selected_user}",
+                version="2023-05-29~beta",
+            )
+            user = user_response.json()
+            user_data = user.get("data", {})
+            user_attributes = user_data.get("attributes", {})
+            return User(
+                id=self.importingUserId,
+                name=user_attributes.get("name"),
+                username=user_attributes.get("username"),
+                email=user_attributes.get("email"),
+            )
+        elif item == "browseUrl":
+            # Ensure that our browse URL matches the tenant the user is making a request to
+            if self.organization.client.api_url.startswith("https://api.eu.snyk.io"):
+                url_prefix = "https://app.eu.snyk.io"
+            elif self.organization.client.api_url.startswith("https://api.au.snyk.io"):
+                url_prefix = "https://app.au.snyk.io"
+            else:
+                url_prefix = "https://app.snyk.io"
+            return f"{url_prefix}/org/{self.organization.slug}/project/{self.id}"
+        else:
+            raise AttributeError(
+                f"'{type(self).__name__}' object has no attribute '{item}'"
```

Comment:
is there a way to do this regex-based so we don't have to handle things like eu and au specifically?  For single tenant it could be any customer identifier e.g. https://api.{customer}.snyk.io so I'm thinking we'd have pattern match extract the {customer} part and then use it to set the url_prefix.  @lili2311 thoughts?